src/parser/Statement.ts

```diff
@@ -3415,3 +3415,68 @@ export class ContinueStatement extends Statement {
         return this.tokens.continue?.leadingTrivia ?? [];
     }
 }
+
+
+export class TypeCastMStatement extends Statement {
+    constructor(options: {
+        typeCast?: Token;
+        as?: Token;
+        m?: Token;
+        typeExpression: TypeExpression;
+    }
+    ) {
+        super();
+        this.tokens = {
+            typeCast: options.typeCast,
+            as: options.as,
+            m: options.m
+        };
+        this.typeExpression = options.typeExpression;
+        this.range = util.createBoundingRange(
+            this.tokens.typeCast,
+            this.tokens.m,
+            this.tokens.as,
+            this.typeExpression
+        );
+    }
+
+    public readonly tokens: {
+        readonly typeCast?: Token;
+        readonly m?: Token;
```

Comment:
Let's make this an `Expression`. In this PR we only support `VariableExpression`, but it could expand to others the future so let's keep it as broad as possible to reduce the chance of breaking changes (or needless duplicate AST nodes). 

Not sure what to name the thing. `item`? `element`? `subject`? `obj`?
```suggestion
        readonly item?: Expression;
```

Reply:
Using `TypecastExpression` here, like I always should have.